core/templates/pages/learner-dashboard-page/new-learner-dashboard/community-lessons-tab.component.ts

```diff
@@ -0,0 +1,309 @@
+  noCommunityLessonActivity: boolean;
+  noPlaylistActivity: boolean;
+  totalIncompleteLessonsList: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  totalCompletedLessonsList: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  totalLessonsInPlaylist: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  allCommunityLessons: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  displayIncompleteLessonsList: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  displayCompletedLessonsList: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  displayLessonsInPlaylist: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  displayInCommunityLessons: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  selectedSection: string;
+  dropdownEnabled: boolean;
+  showMore = {
+    incomplete: false,
+    completed: false,
+    playlist: false,
+    subscriptions: false
+  };
+  pageNumberInCommunityLessons: number = 1;
+  pageSize: number = 3;
+  startIndexInCommunityLessons: number = 0;
+  endIndexInCommunityLessons: number = 3;
+  pageNumberInPlaylist: number = 1;
+  startIndexInPlaylist: number = 0;
+  endIndexInPlaylist: number = 3;
+
+  ngOnInit(): void {
+    this.noCommunityLessonActivity = (
+      (this.incompleteExplorationsList.length === 0) &&
+        (this.completedExplorationsList.length === 0) &&
+        (this.incompleteCollectionsList.length === 0) &&
+        (this.completedCollectionsList.length === 0));
+    this.noPlaylistActivity = (
+      (this.explorationPlaylist.length === 0) &&
+      (this.collectionPlaylist.length === 0));
+    this.totalIncompleteLessonsList.push(
+      ...this.incompleteExplorationsList, ...this.incompleteCollectionsList);
+    this.totalCompletedLessonsList.push(
+      ...this.completedExplorationsList, ...this.completedCollectionsList);
+    this.totalLessonsInPlaylist.push(
+      ...this.explorationPlaylist, ...this.collectionPlaylist);
+    this.allCommunityLessons.push(
+      ...this.incompleteExplorationsList, ...this.incompleteCollectionsList,
+      ...this.completedCollectionsList, ...this.completedExplorationsList);
+    this.displayIncompleteLessonsList = this.totalIncompleteLessonsList.slice(
+      0, 3);
+    this.displayCompletedLessonsList = this.totalCompletedLessonsList.slice(
+      0, 3);
+    if (this.checkMobileView()) {
+      this.displayLessonsInPlaylist = this.totalLessonsInPlaylist;
+    } else {
+      this.displayLessonsInPlaylist = this.totalLessonsInPlaylist.slice(0, 3);
+    }
+    this.displayInCommunityLessons = this.allCommunityLessons;
+    this.selectedSection = 'All';
+    this.dropdownEnabled = false;
+  }
+
+  decodePngURIData(base64ImageData: string): string {
+    return decodeURIComponent(base64ImageData);
+  }
+
+  checkMobileView(): boolean {
+    return this.deviceInfoService.isMobileDevice();
+  }
+
+  enableDropdown(): void {
```

Comment:
Is this used for toggling or just enabling? If just enabling, you can directly assign `true`, right? Else, rename function.

Reply:
Done.